### PR TITLE
Allow asynchronous upload of documents to verification provider

### PR DESCRIPTION
**Pull request by saalistaja**
- Allowed async upload and data extraction in the used document provider
- Introduced a regular task to check process of document upload and data extraction at the provider side
- Added scheduler lock support to prevent parallel run of regular tasks
- Adapted and fixed unit tests to latest changes

### CODE REVIEW

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/database/DocumentResultRepository.java [hunk 1]
```diff
@@ -30,4 +33,13 @@
 @Repository
 public interface DocumentResultRepository extends CrudRepository<DocumentResultEntity, Long> {
 
+    /**
+     * @return All not finished document uploads (in progress status and no extracted data filled)
+     */
+    @Query("select doc from DocumentResultEntity doc where" +
```
Comment (romanstrobl) [inline enrollment-server/src/main/java/com/wultra/app/enrollmentserver/database/DocumentResultRepository.java:39]: Perhaps we should capitalize SQL commands like in other queries?

File: enrollment-server/src/test/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProviderTest.java [hunk 2]
```diff
@@ -47,7 +52,7 @@
  * @author Lukas Lukovsky, lukas.lukovsky@wultra.com
  */
 @SpringBootTest(classes = EnrollmentServerTestApplication.class)
-@ActiveProfiles("external-service")
+@ActiveProfiles({"external-service"})
```
Comment (romanstrobl) [inline enrollment-server/src/test/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProviderTest.java:55]: Is this change required?

**Review by romanstrobl — APPROVED** [review]
Nice addition, I found several minor issues.

File: enrollment-server/src/test/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProviderTest.java (hunk 2 above)
Comment (saalistaja) [reply enrollment-server/src/test/java/com/wultra/app/docverify/zenid/provider/ZenidDocumentVerificationProviderTest.java]: aah, thanks
I had two profiles there and removed the other, no need for an array any more

File: enrollment-server/src/main/java/com/wultra/app/enrollmentserver/database/DocumentResultRepository.java (hunk 1 above)
Comment (saalistaja) [reply enrollment-server/src/main/java/com/wultra/app/enrollmentserver/database/DocumentResultRepository.java]: fixed
